docs/deploy/scrape.py

```diff
@@ -229,15 +232,37 @@ def upload(app_id, api_key, records, version):
     # Pick some settings for this index.
     index.set_settings(SETTINGS)
 
-    # Upload to temp index.
-    print(f"uploading {len(records)} records to temp index {temp_name}...", file=sys.stderr)
-    index.save_objects(records, {"autoGenerateObjectIDIfNotExist": True})
-    print("upload complete", file=sys.stderr)
-
-    # Rename index into place.
-    print(f"renaming temp index {temp_name} -> {final_name}...", file=sys.stderr)
-    client.move_index(temp_name, final_name).wait()
-    print("rename done", file=sys.stderr)
+    num_tries = 1
+    while num_tries <= max_retries:
+        # Upload to temp index.
+        print(f"uploading {len(records)} records to temp index {temp_name}...", file=sys.stderr)
+        index_response = index.save_objects(records, {"autoGenerateObjectIDIfNotExist": True})
+        index_response.wait()
+        print("upload complete", file=sys.stderr)
+
+        # Rename index into place.
+        print(f"renaming temp index {temp_name} -> {final_name}...", file=sys.stderr)
+        client_response = client.move_index(temp_name, final_name)
+        client_response.wait()
+        print("rename done", file=sys.stderr)
+
+        # Verify remote record length
+        print(f"checking that {final_name} contains {len(records)} records")
+        final_index = client.init_index(final_name)
+        search_iterator = final_index.browse_objects()
+        search_iterator.next()
+        remote_length = search_iterator._raw_response["nbHits"]
+        if remote_length == len(records):
+            print(f"verified that {final_name} contains {len(records)} records", file=sys.stderr)
+            break
+        else:
+            print(
+                f"{final_name} contains {remote_length} records but expected {len(records)}"
+                f" records.",
+                file=sys.stderr,
+            )
+            if num_tries == max_retries:
+                raise Exception("Maximum number of retries reached with no success")
```

Comment:
Something other than bare Exception?  `ValueError` or `RuntimeError` are almost as meaningless, but raising `Exception` makes it really hard to catch it except by catching basically everything.  That's why raising `Exception` is generally avoided.